### PR TITLE
dkp: git lfs support added, updated changed_files logic for central pipeline

### PR DESCRIPTION
**Pull request by avinashtrivedi11**
- Using gitlab APIs to get last successful pipeline id and get all changed files till that commit
-  Added git lfs in pipeline for central repo

### CODE REVIEW

File: knowledge-portal/central/initiate-push-to-source.sh [hunk 1]
```diff
@@ -6,15 +6,21 @@ set -e
 git config --global user.email "PlatformTeam@detecttechnologies.com"
 git config --global user.name "Detect Gitlab Bot"
 
-# changed files can be obtained by git diff between latest commit and unstaged area
-git reset --soft HEAD~1
-changed_files=$(git diff --name-only HEAD)
+# Install git lfs
+apt-get update -qq && apt-get install -y -qq git-lfs
```
Comment (ashwinr64) [inline knowledge-portal/central/initiate-push-to-source.sh:10]: You can move this to before_script

File: knowledge-portal/central/initiate-push-to-source.sh [hunk 2]
```diff
@@ -6,15 +6,21 @@ set -e
...
+git lfs install
+git lfs track "*.jpg" "*.jpeg" "*.png" "*.gif" "*.svg" "*.ico" "*.mp3" "*.mp4"
```
Comment (ashwinr64) [inline knowledge-portal/central/initiate-push-to-source.sh:12]: Is it really needed to run git lfs install and track on every commit? Tracking would persist in the .gitattribute file.

File: knowledge-portal/central/initiate-push-to-source.sh [hunk 3]
```diff
@@ -6,15 +6,21 @@ set -e
...
+
+# changed files are all files since pipeline successfully ran last time on central
+
+GITLAB_API="https://gitlab.com/api/v4"
+PIPELINES=$(curl --header "PRIVATE-TOKEN: $PIPELINE_ID_TOKEN" "$GITLAB_API/projects/$CI_PROJECT_ID/pipelines?scope=finished&status=success" | jq '.[0].sha')
+LAST_PIPELINE_COMMIT=$(echo $PIPELINES | tr -d '"')
+changed_files=$(git diff --name-only $LAST_PIPELINE_COMMIT $CI_COMMIT_SHA)
```
Comment (ashwinr64) [inline knowledge-portal/central/initiate-push-to-source.sh:19]: Please add this token generation part to the Usage section and possibly the other tokens/keys that have to be maintained.

File: knowledge-portal/source/initiate-push-to-central.sh [hunk 4]
```diff
@@ -7,6 +7,12 @@ git config --global user.email "PlatformTeam@detecttechnologies.com"
 git config --global user.name "Detect Gitlab Bot"
 git clone $CENTRAL_GIT_PUSH_URL /root/central
 
+# Install git lfs
+cd /root/central
+apt-get update -qq && apt-get install -y -qq git-lfs
+git lfs install
+git lfs track "*.jpg" "*.jpeg" "*.png" "*.gif" "*.svg" "*.ico" "*.mp3" "*.mp4"
```
Comment (ashwinr64) [inline knowledge-portal/source/initiate-push-to-central.sh:14]: Is it really needed to run git lfs install and track on every commit? Tracking would persist in the .gitattribute file.

Comment (avinashtrivedi11) [reply knowledge-portal/source/initiate-push-to-central.sh]: You're right, removed those lines.

File: knowledge-portal/central/initiate-push-to-source.sh (hunk 3 above)
Comment (avinashtrivedi11) [reply knowledge-portal/central/initiate-push-to-source.sh:13]: modified README accordingly

**Pull request merged**